alpha/apps/kaltura/lib/storage/urlTokenizers/kWowzaSecureTokenUrlTokenizer.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+class kWowzaSecureTokenUrlTokenizer extends kUrlTokenizer
+{
+	/**
+	 * @var string
+	 */
+	protected $paramPrefix;
+	
+	/**
+	 * @var string
+	 */
+	protected $hashAlgorithm;
+	
+	public function tokenizeSingleUrl($url, $urlPrefix = null)
+	{
+		$tokenPrefix = $this->getParamPrefix();
+		
+		$expiryTimeFrame = $this->getWindow();
+		
+		$params = array();
+		
+		$params['starttime'] = time();
+		$params['endtime'] = time() + $expiryTimeFrame;
+		
+		$urlTokenParameters = $this->prefixParams($tokenPrefix,$params);
+		
+		$urlTokenParameters[$tokenPrefix.'hash'] = $this->getHash($urlTokenParameters,$url);
+		
+		return $url."?".http_build_query($urlTokenParameters, '', '&');
+	}
+	
+	/**
+	 * Get prefixed parameters
+	 * 
+	 * @return array
+	 */
+	private function prefixParams($prefix, $rawParameters)
+	{
+		$prefixedParameters = array();
+		
+		if(empty($rawParameters))
```

Comment:
Looking at this now, we can remove this if, it will never by empty, and even if it will, the result will be correct